### DISCUSSION TITLE
Archive of `Shield` training videos!

### DISCUSSION THREAD

**Discussion by datamweb** [codeigniter4/shield#446] [General]
We are all trying to make shield users use it in their projects without confusion and difficulty, this thread is created for this reason.
Please submit any training video related to **Shield** in this section:

**Comment by datamweb** [comment]
**1. MASTERING SHIELD (Lonnie Ezell) https://learncodeigniter.net/blog/c/mastering-shield**
2. CodeIgniter Shield Installation and Overview https://www.youtube.com/watch?v=Qp3dX1rD46A
3. Install Codeigniter 4 Authentication with Shield https://www.youtube.com/watch?v=75y3ek_zb_o
4. Codeigniter 4 Api Authentication using Shield https://www.youtube.com/watch?v=dgE5gvA70zc
5. Step by Step RESTful Web APIs Development Using CodeIgniter 4 and Shield Authentication *(**NOT FREE**)* https://www.udemy.com/course/codeigniter-4-apis-development-using-shield-authentication
6. CodeIgniter 4 Shield Authentication and Authorization and Login Registration Shield https://youtu.be/2rGkPv7YVnA

**Reply by mouaad19** [reply]
Please can add a video how can store groups and permissions list in database and manage them with ui interface, actually I have a difficult with : 
Store and Get list of groups and permissions.
assign permissions to groups with UI interface. 

many thanks

**Reply by mouaad19** [reply]
Can you put your request in english so we can help you ?!

**Reply by grisinformatica** [reply]
> PERO no entiendo que simboliza beta.access por ejemplo beta= a un controlador?? o en realidad a que se refiere con Beta??

Hola Javier, Un permiso es simplemente una cadena compuesta por un ámbito (scope) y una acción, separadas por puntos [[Leer aquí]](https://github.com/codeigniter4/shield/blob/develop/docs/authorization.md). Por ejemplo para el ámbito de Noticias podrías tener los permisos `news.create` para crear determinar si un usuario puede crear una noticia o `news.delete` para determinar si el usuario puede borrar o no una noticia.

En el ejemplo se especifica un permiso `beta.access` para determinar si el usuario puede acceder o no a funciones o características que estén en una versión beta, simplemente eso.

**Reply by grisinformatica** [reply]
No, es simplemente una convención para los nombres, no están vinculados a los controladores. Deberás implementar la verificación dentro del controlador o la vista. [Estas funciones](https://github.com/codeigniter4/shield/blob/develop/docs/authorization.md#authorizing-users) pueden ayudarte.

**Comment by carlkyo** [comment]
Thanks Sir.
I would be grateful if you could make some tutorials below

1. user roles and permissions management.
2. how to apply permissions to the routes.
Thanks again for the awesome framework and your efforts

**Reply by datamweb** [reply]
> 2. how to apply permissions to the routes.

```php
$routes->get('users/profile', 'UserController::index', ['filter' => 'permission:admin.access']);
```
@carlkyo more info see : https://github.com/codeigniter4/shield/blob/develop/docs/authorization.md#authorizing-via-routes
